sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobAPITest.groovy

```diff
@@ -115,57 +112,58 @@ class BlockBlobAPITest extends APISpec {
 
     def "Stage block error"() {
         setup:
-        bu = cu.getBlockBlobClient(generateBlobName())
+        bc = cc.getBlockBlobClient(generateBlobName())
 
         when:
-        bu.stageBlock("id", defaultInputStream.get(), defaultDataSize)
+        bc.stageBlock("id", defaultInputStream.get(), defaultDataSize)
 
         then:
         thrown(StorageException)
     }
 
     def "Stage block from url"() {
         setup:
-        cu.setAccessPolicy(PublicAccessType.CONTAINER, null)
-        def bu2 = cu.getBlockBlobClient(generateBlobName())
+        cc.setAccessPolicy(PublicAccessType.CONTAINER, null)
+        def bu2 = cc.getBlockBlobClient(generateBlobName())
         def blockID = getBlockID()
 
         when:
-        HttpHeaders headers = bu2.stageBlockFromURLWithResponse(blockID, bu.getBlobUrl(), null, null, null, null, null, null).headers()
-        Iterator<BlockItem> listResponse = bu2.listBlocks(BlockListType.ALL).iterator()
-        BlockItem block = listResponse.next()
-        bu2.commitBlockList(Arrays.asList(blockID))
-        ByteArrayOutputStream outputStream = new ByteArrayOutputStream()
-        bu2.download(outputStream)
+        def headers = bu2.stageBlockFromURLWithResponse(blockID, bc.getBlobUrl(), null, null, null, null, null, null).headers()
 
         then:
         headers.value("x-ms-request-id") != null
         headers.value("x-ms-version") != null
         headers.value("x-ms-content-crc64") != null
         headers.value("x-ms-request-server-encrypted") != null
 
+        def response = bu2.listBlocks(BlockListType.ALL)
+        response.uncommittedBlocks().size() == 1
```

Comment:
What was the conversation around spliting up these two lists? It seems like we'd want to be consistent with the fact that blobItems and prefixes are presented in one list

Reply:
My understanding is this:

For listing blobs in a container, this is a paginated listing operation that returns two different lists, so auto-continuing two lists at the same time while someone is likely only consuming one at a time is a problematic approach. Therefore we join the lists and have a flag to tell you what is what. Since this API simulates listing blobs and "directories," this is actually useful, as the JDK provides similar functionality when browsing the local filesystem.

For listing blocks in a blob, this is not paginated. In the world where we are not using the PagedFlux listing approach for operations that are not paged, we are already returning a Mono of a buffered data structure, and so we can keep the two lists pre-separated for the user.

Reply:
@JonathanGiles may have input on this.

Reply:
Answer to this is likely dependent on outcome of #5097